pages/content/migration-guidance.mdx

```diff
@@ -373,6 +374,56 @@ Principles](https://www.ncsc.gov.uk/collection/cloud/the-cloud-security-principl
+
+
+For example:
+- [GOV.UK Notify technical documentation](https://github.com/alphagov/notifications-tech-docs)
+- [GOV.Pay technical  documentation](https://github.com/alphagov/pay-tech-docs)  
+
+
+The [decommission of GOV.UK PaaS](https://gds.blog.gov.uk/2022/07/12/why-weve-decided-to-decommission-gov-uk-paas-platform-as-a-service/) requires that these websites are migrated. 
+
+Teams need to decide on an appropriate alternative. This guidance documents the approach taken by the PaaS team in the hope that it will prove useful to other teams. 
+
+A common pattern uses [GitHub Actions](https://docs.github.com/en/actions) to publish the content to [GitHub Pages](https://pages.github.com/).
+
+The GOV.UK PaaS team used this approach to move our technical documentation from our own Cloud Foundry based platform to GitHub Pages. 
+
+The content is generated by the [Middleman](https://middlemanapp.com/) static site publishing tool from a set of templates defined in Markdown and [Ruby ERBs](https://github.com/ruby/erb) however the approach can be adapted for use with other static publishing tools like [Jekyll](https://jekyllrb.com/) or [Hugo](https://gohugo.io/) by modifying the pipeline to replace the step that generates the HTML content.
+
+Using the [PaaS tech docs](https://docs.cloud.service.gov.uk) as a concrete example:
+
+- source code [alphagov/paas-tech-docs](https://github.com/alphagov/paas-tech-docs) repository
+- published to [alphagov.github.io/paas-tech-docs](https://alphagov.github.io/paas-tech-docs)
+- custom domain [docs.cloud.service.gov.uk](https://docs.cloud.service.gov.uk) to preserve continuity for our users
+
+The GitHub [`deploy.yml`](https://github.com/alphagov/paas-tech-docs/blob/main/.github/workflows/deploy.yml) workflow performs the following steps when changes are made to the [`main`](https://github.com/alphagov/paas-tech-docs/tree/main) branch.
+
+- check out the repository source code using [actions/checkout](https://github.com/marketplace/actions/checkout) to work with
+- set up Nodejs tools using [actions/setup-node](https://github.com/marketplace/actions/setup-node-js-environment)
+- set up Ruby tools using [actions/setup-ruby](https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby)
+- set up Python tools using [actions/setup-python](https://github.com/marketplace/actions/setup-python)
+- install Python and Nodejs dependencies
+- run tests in Nodejs
+- run tests in Python
+- run Middleman using `bundle exec middleman build` to generate static content in the `/build` directory
+- create an artifact from the generated `/build` directory using the [actions/upload-github-pages-artifact](https://github.com/marketplace/actions/upload-github-pages-artifact) storing the content in a gzip archive called `github-pages`
+- publish to GitHub pages using the [actions/deploy-pages](https://github.com/actions/deploy-pages)
+- notify the PaaS team via Slack if there are any problems
```

Comment:
do we need to explain what each action does?

Reply:
yes